C++/C.sublime-syntax

```diff
@@ -13,1427 +14,733 @@ first_line_match: |-
-          [,;:_]?                                               # separator character (AltiVec)
-          ((-?\d+)|\*(-?\d+\$)?)?                               # minimum field width
-          (\.((-?\d+)|\*(-?\d+\$)?)?)?                          # precision
-          (hh|h|ll|l|j|t|z|q|L|H|D|DD|vh|vl|v|hv|hl|wf?[0-9]+)? # length modifier
-          (\[[^\]]+\]|[am]s|[diouxXbDOUeEfFgGaACcSspn%])        # conversion type
+          (\d+\$)?                                      # field (argument #)
+          [#0\- +']*                                    # flags
+          [,;:_]?                                       # separator character (AltiVec)
+          ((-?\d+)|\*(-?\d+\$)?)?                       # minimum field width
+          (\.((-?\d+)|\*(-?\d+\$)?)?)?                  # precision
+          (hh|h|ll|l|j|t|z|q|L|vh|vl|v|hv|hl)?          # length modifier
+          (\[[^\]]+\]|[am]s|[diouxXDOUeEfFgGaACcSspn%]) # conversion type
       scope: constant.other.placeholder.c
 
-  keywords:
-    - match: \bbreak\b
-      scope: keyword.control.flow.break.c
-    - match: \bcontinue\b
-      scope: keyword.control.flow.continue.c
-    - match: \bgoto\b
-      scope: keyword.control.flow.goto.c
-    - match: \breturn\b
-      scope: keyword.control.flow.return.c
-    - match: \b({{control_keywords}})\b
-      scope: keyword.control.c
-    - match: \b(sizeof|alignof|_Alignof|static_assert|_Static_assert)\b
+  operator:
+    - include: operator.accessor
+    - include: operator.assignment
+    - include: operator.arithmetic
+    - include: operator.logical
+    - include: operator.bitwise
+    - include: operator.comparison
+    - include: operator.word
+    - include: operator.ternary
+    - include: operator.variadic
+    - include: operator.increment-decrement
+  operator.assignment:
+    - match: =
+      scope: keyword.operator.assignment.c
+    - match: \+=|-=|\*=|/=|%=|&=|\|=|\^=|<<=|>>=|>>=
+      scope: keyword.operator.assignment.augmented.c
+  operator.arithmetic:
+    - match: \+|-|\*|/|%|!
+      scope: keyword.operator.arithmetic.c
+  operator.logical:
+    - match: \&\&|\|\|
+      scope: keyword.operator.logical.c
+  operator.bitwise:
+    - match: \&|\||\^|~|<<|>>
+      scope: keyword.operator.bitwise.c
+  operator.comparison:
+    - match: ==|!=|<=|>=|<|>
+      scope: keyword.operator.comparison.c
+  operator.word:
+    - match: \b(?:sizeof|_Alignof|alignof|offsetof|_Static_assert|static_assert|_Pragma)\b
       scope: keyword.operator.word.c
+  operator.ternary:
+    - match: '\?|\:'
+      scope: keyword.operator.ternary.c
+  operator.variadic:
+    - match: \.\.\.
+      scope: keyword.operator.variadic.c
+  operator.increment-decrement:
+    - match: \+\+|--
+      scope: keyword.operator.arithmetic.c
+  operator.accessor:
+    - match: \.|\-\>
+      scope: punctuation.accessor.c
+  operator.pointer:
+    - match: \&|\*
+      scope: keyword.operator.arithmetic.c
```

Comment:
Is this the right scope? I think the old tests had `keyword.operator.arithmetic` just because they couldn't identify that these were pointers.